query/src/main/java/org/infinispan/search/mapper/search/loading/context/impl/InfinispanLoadingContext.java

```diff
@@ -62,34 +57,34 @@ public InfinispanLoadingContext build() {
       }
    }
 
-   private class InfinispanSelectionLoadingStrategy<E> implements PojoSelectionLoadingStrategy<E> {
-
-      private final PojoSelectionEntityLoader<E> entityLoader;
-
-      private InfinispanSelectionLoadingStrategy(PojoSelectionEntityLoader<E> entityLoader) {
-         this.entityLoader = entityLoader;
-      }
-
-      @Override
-      public boolean equals(Object o) {
-         if (this == o) {
-            return true;
-         }
-         if (o == null || getClass() != o.getClass()) {
-            return false;
-         }
-         InfinispanSelectionLoadingStrategy<?> that = (InfinispanSelectionLoadingStrategy<?>) o;
-         return entityLoader.equals(that.entityLoader);
-      }
+//   private class InfinispanSelectionLoadingStrategy<E> implements PojoSelectionLoadingStrategy<E> {
+//
+//      private final PojoSelectionEntityLoader<E> entityLoader;
+//
+//      private InfinispanSelectionLoadingStrategy(PojoSelectionEntityLoader<E> entityLoader) {
+//         this.entityLoader = entityLoader;
+//      }
+//
+//      @Override
+//      public boolean equals(Object o) {
+//         if (this == o) {
+//            return true;
+//         }
+//         if (o == null || getClass() != o.getClass()) {
+//            return false;
+//         }
+//         InfinispanSelectionLoadingStrategy<?> that = (InfinispanSelectionLoadingStrategy<?>) o;
+//         return entityLoader.equals(that.entityLoader);
+//      }
 
       @Override
       public int hashCode() {
          return entityLoader.hashCode();
       }
 
-      @Override
-      public PojoSelectionEntityLoader<E> createLoader(Set<? extends PojoLoadingTypeContext<? extends E>> expectedTypes) {
-         return entityLoader;
-      }
-   }
+//      @Override
+//      public PojoSelectionEntityLoader<E> createLoader(Set<? extends PojoLoadingTypeContext<? extends E>> expectedTypes) {
+//         return entityLoader;
+//      }
+//   }
```

Comment:
Should be reimplemented to take the entity loader from the context passed to `PojoSelectionLoadingStrategy#createEntityLoader`.

See https://github.com/hibernate/hibernate-search/pull/3954/commits/14b61ecf4fb972712528b9fbb889a1a5eec6e92d#diff-57a27b5a52b39a46a54c7f8823c0d7e9e387b1ec806ea67fea83313526936532R51